pallets/erc20-xcm-bridge/src/lib.rs

```diff
@@ -65,14 +65,19 @@ pub mod pallet {
 		pub fn is_erc20_asset(asset: &MultiAsset) -> bool {
 			Erc20Matcher::<T::Erc20MultilocationPrefix>::is_erc20_asset(asset)
 		}
-		pub fn weight_of_erc20_transfer() -> Weight {
-			T::GasWeightMapping::gas_to_weight(T::Erc20TransferGasLimit::get(), true)
+		pub fn gas_limit_of_erc20_transfer(asset: &MultiAsset) -> u64 {
```

Comment:
This function is useless and increase the call stack for nothing, please remove it and move it's body inside weight_of_erc20_transfer

Reply:
Why is the function useless? We need to get the gas limit (not the weight) [here](https://github.com/moonbeam-foundation/moonbeam/pull/2422/files#diff-60c41c4cef32c59e32153a0cdac0d3c4765c302f0b51e28d1fe7afc0b949f1bbR162) and [here](https://github.com/moonbeam-foundation/moonbeam/pull/2422/files#diff-60c41c4cef32c59e32153a0cdac0d3c4765c302f0b51e28d1fe7afc0b949f1bbR212) for the `erc20_transfer` call, right?

Reply:
Indeed, my bad